frontend/src/framework/utils/atomUtils.ts

```diff
@@ -57,3 +58,54 @@ export function atomWithQueries<
         return results as TCombinedResult;
     });
 }
+
+/**
+ * Creates an writeable atom that persists it's value in localstorage, but keeps it unique per module instance. **Note: must only be used within a module instances own store**
+ * @param storageKey Prefix for the key. Should be unique among a module's atoms. Actual storage key will be: `<prefix>:<moduleInstanceId>`
+ * @param initialValue Initial atom value
+ * @returns A get/set atom
+ */
+export function atomWithModuleInstanceStorage<T>(storageKey: string, initialValue: T) {
+    // simple atom, needed to trigger new lookups
+    const instanceValueAtom = atom<T | null>(null);
+
+    return atom<T, [T], void>(
+        (get) => {
+            const instanceId = getIntanceId(get);
+            const fullStorageKey = makeInstanceStorageKey(instanceId, storageKey);
+            const instanceValue = get(instanceValueAtom);
+            const storedValue = localStorage.getItem(fullStorageKey);
+
+            if (instanceValue !== null) return instanceValue;
+            else if (storedValue !== null) return JSON.parse(storedValue);
+            else return initialValue;
+        },
+        (get, set, newVal: T) => {
+            const instanceId = getIntanceId(get);
+            const fullStorageKey = makeInstanceStorageKey(instanceId, storageKey);
+
+            localStorage.setItem(fullStorageKey, JSON.stringify(newVal));
+            set(instanceValueAtom, newVal);
+        }
+    );
+}
+
+export function clearModuleInstanceStorage(instanceId: string, key: string) {
+    const fullKey = makeInstanceStorageKey(instanceId, key);
+    localStorage.removeItem(fullKey);
+}
+
+function makeInstanceStorageKey(instanceId: string, key: string): string {
+    // ! Be mindful about changing this; existing stored values will dissapear!
+    return `${key}:${instanceId}`;
+}
+
+function getIntanceId(get: Getter): string {
+    const id = get(CurrentModuleInstanceIdAtom);
+
+    if (id === null) {
+        throw new Error("Module instance not set. Make sure this atom is only used within a module storage");
+    }
+
+    return id;
+}
```

Comment:
Thats a fair concern. Im fine with closing down the PR and then we can come back to it later. Would it be okay to use this persistence solution privately inside the well-log viewer module still?

If we close the PR for now; since you noted that the `onInstanceUnload` would be nice, I could open a new PR that only introduces that logic by itself if wanted

Reply:
Given the fact that we are going to implement the general persistence mechanism, but not before Q1 2025, I would vote for keeping your mechanism as a private module implementation. I think it's very beneficial that the `WellLogViewer` is storing its status until we come up with the general solution. And by keeping it private (in the module folder), we avoid the risk of others starting to use it.